File: leetcode/171/Main.rb
# @param {String} column_title
# @return {Integer}
def title_to_number(column_title)
    sum = 0;

    # column_title.chars.each do |c|
    column_title.each_byte do |c|
        sum = sum *26 + c -65 +1
    end

    sum
end


puts title_to_number("A")
puts title_to_number("AB")
puts title_to_number("ZY")

File: leetcode/171/Main.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function(columnTitle) {
    let sum = 0;

    for(let i = 0; i < columnTitle.length;i++)
    {
        sum = sum*26 + columnTitle.charCodeAt(i) -"A".charCodeAt(0)+1;
    }
    return sum
};

console.log(titleToNumber("A"));
console.log(titleToNumber("AB"));
console.log(titleToNumber("ZY"));